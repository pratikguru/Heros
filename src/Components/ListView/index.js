import React, { Component } from "react";
import Animations from "../../Utils/animations.js";

import {
  TableContainer,
  TableHeader,
  Header,
  TableBody,
  TableRowsDesktop,
  TableRowsMobile,
  TableCells,
  ImageHolder,
  ImageDescriptor
} from "./listViewStyle.js";

const container = {
  hidden: { opacity: 1, scale: 0 },
  visible: {
    opacity: 1,
    scale: 1,
    transition: {
      delay: 0.1,
      when: "beforeChildren",
      staggerChildren: 0.1
    }
  }
};

const item = {
  hidden: { y: 20, opacity: 0 },
  visible: {
    y: 0,
    opacity: 1
  }
};

export default class ListView extends Component {
  rowClickHandler = index => {
    this.props.handleOnClick(index);
  };

  render() {
    return (
      <TableContainer>
        <TableHeader>
          <Header>Heroes</Header>
          <Header>Type</Header>
          <Header>Description</Header>
        </TableHeader>

        <TableBody variants={container} initial="hidden" animate="visible">
          {this.props.heroes.map((value, index) => (
            <div key={index}>
              <TableRowsDesktop
                key={index}
                onClick={() => this.rowClickHandler(value.id)}
                variants={item}
                whileHover={Animations.rowClick.whileHover}
                whileTap={Animations.rowClick.whileTap}
              >
                <TableCells>
                  <ImageHolder src={value.avatar_url}></ImageHolder>
                  <ImageDescriptor>{value.full_name}</ImageDescriptor>
                </TableCells>
                <TableCells>{value.type.name}</TableCells>
                <TableCells>{value.description}</TableCells>
              </TableRowsDesktop>
              <TableRowsMobile
                onClick={() => this.rowClickHandler(value.id)}
                whileTap={{ scale: 0.98 }}
              >
                <div
                  style={{
                    width: "100%",
                    display: "flex",
                    flexDirection: "row",
                    justifyContent: "flex-start",
                    padding: "10px",
                    paddingBottom: "10px"
                  }}
                >
                  <ImageHolder src={value.avatar_url}></ImageHolder>
                  <div>
                    <ImageDescriptor>{value.full_name}</ImageDescriptor>
                    <div>{value.type.name}</div>
                  </div>
                </div>
                <TableCells>{value.description}</TableCells>
              </TableRowsMobile>
            </div>
          ))}
        </TableBody>
      </TableContainer>
    );
  }
}
