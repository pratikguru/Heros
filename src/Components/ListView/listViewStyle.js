import styled from "styled-components";
import media from "../../Utils/media.js";
import { motion } from "framer-motion";
export const TableContainer = styled.div`
  width: 90%;
  height: 80vh;
  display: flex;
  flex-direction: column;
`;

export const TableHeader = styled.div`
  width: 100%;
  height: auto;
  padding: 1px;
  display: flex;
`;

export const Header = styled.div`
  display: flex;
  width: 100%;
  height: auto;
  padding: 2px;
  font-weight: bold;
  color: grey;
  font-size: 15px;

  ${media.phone`
    display : none;
  `}
`;

export const TableBody = styled(motion.div)`
  width: 100%;
  height: 100%;
  display: block;
  flex-direction: column;
  overflow-y: auto;
  padding: 1px;
  flex-wrap: nowrap;
  background-color: #eaeaea;
  border-radius: 5px;
`;

export const TableRowsDesktop = styled(motion.div)`
  width: auto;
  height: 70px;
  display: flex;
  padding: 2px;
  margin: 4px;
  border-radius: 4px;
  background-color: #f3f3f3;

  ${media.phone`
    display : none;
  `}
`;

export const TableRowsMobile = styled(motion.div)`
  ${media.phone`
    width : auto;
    height : 110px;
    display:  flex;
    margin: 8px;
    padding : 2px;
    border-radius : 4px;
    background-color: #f3f3f3;
    align-items : center;
    flex-direction : column;
    padding-bottom : 10px;
    `}

  display : none;
`;

export const TableCells = styled.div`
  width: fill-available;
  height: auto;
  display: flex;
  align-items: center;
  margin: 2px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  padding: 5px;
`;

export const ImageHolder = styled.img`
  height: 50px;
  width: 50px;
  margin-left: 10px;
  margin-right: 25px;
`;

export const ImageDescriptor = styled.div`
  font-weight: bold;
  font-size: 20px;
  color: black;
`;

TableBody.displayName = "TableBody";
TableCells.displayName = "TableCells";
TableRowsDesktop.displayName = "TableRowDesktop";
