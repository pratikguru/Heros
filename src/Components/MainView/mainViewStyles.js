import styled from "styled-components";
import { motion } from "framer-motion";
import media from "../../Utils/media.js";
export const OuterContainer = styled.div`
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  overflow-x: hidden;
  padding: 10px;
`;

export const Header = styled.div`
  width: 90%;
  display: flex;
  height: auto;
  padding: 10px;
  justify-content: flex-start;

  ${media.phone`
     justify-content: center;
  `}
`;

export const AddButton = styled(motion.div)`
  border-radius: 5px;
  height: 40px;
  width: 120px;
  color: white;
  background-color: #1bb11b;
  display: flex;
  justify-content: center;
  align-items: center;
  box-shadow: none;
  text-selection: none;
  cursor: pointer;

  ${media.phone`
    width : 100%;
  `}
`;

export const LoadingSpinner = styled(motion.div)`
  width: 300px;
  height: 80vh;
  display: flex;
  justify-content: center;
  align-items: center;

  font-size: 20px;
`;

export const SearchField = styled.input`
  width: 250px;
  max-width: 250px;
  border-radius: 6px;
  border: none;
  background-color: #0a08080f;
  height: 40px;
  outline: none;
  padding: 8px;
  margin-left: 10px;
`;
AddButton.displayName = "AddButton";
