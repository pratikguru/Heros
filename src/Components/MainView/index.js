import React, { Component } from "react";

import Animations from "../../Utils/animations.js";
import ListView from "../ListView";
import ViewHeroModal from "../Modal/viewHero.js";
import AddHeroModal from "../Modal/addHero.js";
import {
  OuterContainer,
  Header,
  AddButton,
  LoadingSpinner,
  SearchField
} from "./mainViewStyles.js";
const BackendURL = "http://127.0.0.1:4000";

export default class MainView extends Component {
  constructor() {
    super();
    this.state = {
      mainResponse: [],
      showViewModal: false,
      showAddModal: false,
      selectedHero: {},
      heroTypes: {},
      searchField: "",
      filteredMainResponse: []
    };

    this.fetchAllHeroes = this.fetchAllHeroes.bind(this);
    this.deleteHero = this.deleteHero.bind(this);
    this.addNewHero = this.addNewHero.bind(this);
  }

  componentDidMount() {
    let fetchPromise = fetch(BackendURL + "/heroes/");
    fetchPromise
      .then(response => response.json())
      .then(response => {
        /* Loading all hero types on load. */
        var heroTypes = response.map(value => ({
          id: value.type.id,
          name: value.type.name
        }));

        /* Filtering all unique types.*/
        let uniqueArray = Array.from(
          new Set(heroTypes.map(JSON.stringify))
        ).map(JSON.parse);

        this.setState({
          mainResponse: response,
          heroTypes: uniqueArray
        });
      })
      .catch(err => {
        console.log(err);
      });
  }

  deleteHero(hero) {
    let deletePromise = fetch(BackendURL + "/heroes/" + hero.id, {
      method: "DELETE"
    });

    deletePromise.then(response => {
      if (!response.ok) throw new Error(response.status);
      else return deletePromise;
    });
    return deletePromise;
  }

  addNewHero(avatar) {
    let addPromise = fetch(BackendURL + "/heroes/", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(avatar)
    });

    addPromise.then(response => {
      if (!response.ok) throw new Error(response.status);
      else return addPromise;
    });
    return addPromise;
  }

  fetchAllHeroes() {
    let fetchPromise = fetch(BackendURL + "/heroes/");
    fetchPromise
      .then(response => response.json())
      .then(response => {
        this.setState({
          mainResponse: response
        });
      })
      .catch(err => {
        console.log(err);
      });
    return fetchPromise;
  }

  handleHeroDelete = hero => {
    this.deleteHero(hero)
      .then(() => {
        this.fetchAllHeroes().then(() => {
          this.setState({
            showViewModal: false
          });
        });
      })
      .catch(error => {
        console.log(error);
      });
  };

  handleAddNewAvatar = avatar => {
    this.addNewHero(avatar)
      .then(() => {
        this.fetchAllHeroes().then(() => {
          this.setState({
            showAddModal: false
          });
        });
      })
      .catch(error => {
        console.log(error);
      });
  };

  handleHeroAdd = () => {
    this.setState({
      showAddModal: true
    });
  };

  handleSearch = e => {
    this.setState({
      searchField: e.target.value
    });
  };

  viewAvatar = index => {
    let hero = this.state.mainResponse.filter(function(hero) {
      if (hero.id === index) {
        return hero;
      } else {
        return null;
      }
    });

    this.setState({
      showViewModal: true,
      selectedHero: hero[0]
    });
  };

  render() {
    const { mainResponse, searchField } = this.state;
    const filterdList = mainResponse.filter(response =>
      response.full_name.toLowerCase().includes(searchField.toLowerCase())
    );

    return (
      <OuterContainer>
        <Header>
          <AddButton
            onClick={this.handleHeroAdd}
            whileTap={Animations.buttonClick.whileTap}
            transition={Animations.buttonClick.transition}
          >
            + Add Hero
          </AddButton>
          <SearchField
            type="search"
            placeholder="Search Avatar"
            onChange={this.handleSearch}
          ></SearchField>
        </Header>
        {filterdList.length > 0 ? (
          <>
            <ListView heroes={filterdList} handleOnClick={this.viewAvatar} />
          </>
        ) : (
          <LoadingSpinner
            animate={Animations.loading.animate}
            transition={Animations.loading.transition}
          >
            {" "}
            Loading ...{" "}
          </LoadingSpinner>
        )}
        <ViewHeroModal
          show={this.state.showViewModal}
          selectedHero={this.state.selectedHero}
          handleClose={() => this.setState({ showViewModal: false })}
          handleDelete={this.handleHeroDelete}
        />
        <AddHeroModal
          show={this.state.showAddModal}
          handleClose={() => this.setState({ showAddModal: false })}
          heroTypes={this.state.heroTypes}
          handleAddNewAvatar={this.handleAddNewAvatar}
        />
      </OuterContainer>
    );
  }
}
