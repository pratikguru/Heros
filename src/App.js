import React from "react";
import MainView from "./Components/MainView";

function App() {
  return (
    <div className="App">
      <MainView />
    </div>
  );
}

export default App;
