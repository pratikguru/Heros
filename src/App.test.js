import React from "react";
import { render } from "@testing-library/react";
import App from "./App";
import ListView from "./Components/ListView";
import MainView from "./Components/MainView";
import ViewHeroModal from "./Components/Modal/viewHero";
import ViewAddModal from "./Components/Modal/addHero";

it("Renders List View successfully.", () => {
  const wrapper = render(
    <ListView
      heroes={[
        {
          full_name: "sd",
          avatar_url: "dfdf",
          type: { id: "ddfdf", name: "dfdf" },
          description: "sdsdsD"
        },
        {
          full_name: "sd",
          avatar_url: "dfdf",
          type: { id: "ddfdf", name: "dfdf" },
          description: "sdsdsD"
        }
      ]}
      handleOnClick={() => {}}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it("Renders row contents of list view correctly", () => {
  const wrapper = mount(
    <ListView
      heroes={[
        {
          full_name: "sd",
          avatar_url: "dfdf",
          type: { id: "ddfdf", name: "dfdf" },
          description: "sdsdsD"
        }
      ]}
      handleOnClick={() => {}}
    />
  );

  wrapper
    .find("TableRowDesktop")
    .first()
    .find("TableCells")
    .first()
    .text();
});

it("Listens for view modal click correctly from list view.", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <ListView
      heroes={[
        {
          full_name: "sd",
          avatar_url: "dfdf",
          type: { id: "ddfdf", name: "dfdf" },
          description: "sdsdsD"
        }
      ]}
      handleOnClick={spy}
    />
  );

  wrapper
    .find("TableRowDesktop")
    .first()
    .simulate("click");
  expect(spy.calledOnce).toBe(true);
});

it("Listens for view hero modal to popup.", () => {
  const wrapper = mount(<ViewHeroModal show={true} selectedHero={{}} />);
  const overlay = wrapper.find(".modal-content");
  expect(overlay).toHaveLength(1);
});

it("Listens for view hero modal close click correctly.", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <ViewHeroModal show={true} selectedHero={{}} handleClose={spy} />
  );

  wrapper
    .find(".close-modal")
    .first()
    .simulate("click");

  expect(spy.calledOnce).toBe(true);
});

it("Listens for Add Hero to popup correctly.", () => {
  const wrapper = mount(
    <ViewAddModal show={true} handleClose={() => {}} heroTypes={[]} />
  );
  const overlay = wrapper.find(".modal-content");
  expect(overlay).toHaveLength(1);
});

it("Listens for Add Hero to close correctly.", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <ViewAddModal show={true} handleClose={spy} heroTypes={[]} />
  );

  wrapper
    .find(".close-modal")
    .first()
    .simulate("click");
  expect(spy.calledOnce).toBe(true);
});

it("Listens for Add hero modal to have forms.", () => {
  const wrapper = mount(
    <ViewAddModal
      show={true}
      handleClose={() => {}}
      heroTypes={[{ name: "human" }, { name: "animal" }]}
    />
  );

  expect(wrapper.find("AvatarURLInput").exists()).toBe(true);
  expect(wrapper.find("AvatarNameInput").exists()).toBe(true);
  expect(wrapper.find("AvatarTypeSelect").exists()).toBe(true);
});

it("Checking save button is disabled before filling the form", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <ViewAddModal
      show={true}
      handleClose={spy}
      heroTypes={[{ name: "human" }]}
    />
  );

  wrapper
    .find("AddButton")
    .first()
    .simulate("click");

  console.log(spy.calledOnce);
  expect(spy.calledOnce).toBe(false);
});

it("Check for activation of save button on valid form.", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <ViewAddModal
      show={true}
      heroTypes={[
        { id: "1", name: "human" },
        { id: "2", name: "animal" }
      ]}
      handleAddNewAvatar={spy}
    />
  );

  wrapper.find("AvatarURLInput").simulate("change", {
    target: {
      value: "lol"
    }
  });

  wrapper.find("AvatarNameInput").simulate("change", {
    target: {
      value: "name"
    }
  });

  wrapper.find("AvatarTypeSelect").simulate("change", {
    target: {
      value: "2"
    }
  });

  wrapper.find("TextArea").simulate("change", {
    target: {
      value: "lots of information"
    }
  });

  wrapper
    .find("AddButton")
    .first()
    .simulate("click");

  expect(spy.calledOnce).toBe(true);
});

it("Check for close modal on delete", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <ViewHeroModal
      show={true}
      selectedHero={{
        full_name: "sd",
        avatar_url: "dfdf",
        type: { id: "ddfdf", name: "dfdf" },
        description: "sdsdsD"
      }}
      handleClose={() => {}}
      handleDelete={spy}
    />
  )
    .find("DeleteIconContainer")
    .simulate("click");

  expect(spy.calledOnce).toBe(true);
});
