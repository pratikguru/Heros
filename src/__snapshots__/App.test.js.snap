// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders List View successfully. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm gbbYnw"
      >
        <div
          class="sc-AxirZ cZOeun"
        >
          <div
            class="sc-AxiKw kVuTnU"
          >
            Heroes
          </div>
          <div
            class="sc-AxiKw kVuTnU"
          >
            Type
          </div>
          <div
            class="sc-AxiKw kVuTnU"
          >
            Description
          </div>
        </div>
        <div
          class="sc-AxhCb fqilgR"
          style="opacity: 1; transform: scale(0) translateZ(0);"
        >
          <div>
            <div
              class="sc-AxhUy lakhlt"
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <div
                class="sc-AxheI dAvOVz"
              >
                <img
                  class="sc-Axmtr gWLChO"
                  src="dfdf"
                />
                <div
                  class="sc-AxmLO fHPaRY"
                >
                  sd
                </div>
              </div>
              <div
                class="sc-AxheI dAvOVz"
              >
                dfdf
              </div>
              <div
                class="sc-AxheI dAvOVz"
              >
                sdsdsD
              </div>
            </div>
            <div
              class="sc-AxgMl caBBol"
            >
              <div
                style="width: 100%; display: flex; flex-direction: row; justify-content: flex-start; padding: 10px 10px 10px 10px;"
              >
                <img
                  class="sc-Axmtr gWLChO"
                  src="dfdf"
                />
                <div>
                  <div
                    class="sc-AxmLO fHPaRY"
                  >
                    sd
                  </div>
                  <div>
                    dfdf
                  </div>
                </div>
              </div>
              <div
                class="sc-AxheI dAvOVz"
              >
                sdsdsD
              </div>
            </div>
          </div>
          <div>
            <div
              class="sc-AxhUy lakhlt"
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <div
                class="sc-AxheI dAvOVz"
              >
                <img
                  class="sc-Axmtr gWLChO"
                  src="dfdf"
                />
                <div
                  class="sc-AxmLO fHPaRY"
                >
                  sd
                </div>
              </div>
              <div
                class="sc-AxheI dAvOVz"
              >
                dfdf
              </div>
              <div
                class="sc-AxheI dAvOVz"
              >
                sdsdsD
              </div>
            </div>
            <div
              class="sc-AxgMl caBBol"
            >
              <div
                style="width: 100%; display: flex; flex-direction: row; justify-content: flex-start; padding: 10px 10px 10px 10px;"
              >
                <img
                  class="sc-Axmtr gWLChO"
                  src="dfdf"
                />
                <div>
                  <div
                    class="sc-AxmLO fHPaRY"
                  >
                    sd
                  </div>
                  <div>
                    dfdf
                  </div>
                </div>
              </div>
              <div
                class="sc-AxheI dAvOVz"
              >
                sdsdsD
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm gbbYnw"
    >
      <div
        class="sc-AxirZ cZOeun"
      >
        <div
          class="sc-AxiKw kVuTnU"
        >
          Heroes
        </div>
        <div
          class="sc-AxiKw kVuTnU"
        >
          Type
        </div>
        <div
          class="sc-AxiKw kVuTnU"
        >
          Description
        </div>
      </div>
      <div
        class="sc-AxhCb fqilgR"
        style="opacity: 1; transform: scale(0) translateZ(0);"
      >
        <div>
          <div
            class="sc-AxhUy lakhlt"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            <div
              class="sc-AxheI dAvOVz"
            >
              <img
                class="sc-Axmtr gWLChO"
                src="dfdf"
              />
              <div
                class="sc-AxmLO fHPaRY"
              >
                sd
              </div>
            </div>
            <div
              class="sc-AxheI dAvOVz"
            >
              dfdf
            </div>
            <div
              class="sc-AxheI dAvOVz"
            >
              sdsdsD
            </div>
          </div>
          <div
            class="sc-AxgMl caBBol"
          >
            <div
              style="width: 100%; display: flex; flex-direction: row; justify-content: flex-start; padding: 10px 10px 10px 10px;"
            >
              <img
                class="sc-Axmtr gWLChO"
                src="dfdf"
              />
              <div>
                <div
                  class="sc-AxmLO fHPaRY"
                >
                  sd
                </div>
                <div>
                  dfdf
                </div>
              </div>
            </div>
            <div
              class="sc-AxheI dAvOVz"
            >
              sdsdsD
            </div>
          </div>
        </div>
        <div>
          <div
            class="sc-AxhUy lakhlt"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            <div
              class="sc-AxheI dAvOVz"
            >
              <img
                class="sc-Axmtr gWLChO"
                src="dfdf"
              />
              <div
                class="sc-AxmLO fHPaRY"
              >
                sd
              </div>
            </div>
            <div
              class="sc-AxheI dAvOVz"
            >
              dfdf
            </div>
            <div
              class="sc-AxheI dAvOVz"
            >
              sdsdsD
            </div>
          </div>
          <div
            class="sc-AxgMl caBBol"
          >
            <div
              style="width: 100%; display: flex; flex-direction: row; justify-content: flex-start; padding: 10px 10px 10px 10px;"
            >
              <img
                class="sc-Axmtr gWLChO"
                src="dfdf"
              />
              <div>
                <div
                  class="sc-AxmLO fHPaRY"
                >
                  sd
                </div>
                <div>
                  dfdf
                </div>
              </div>
            </div>
            <div
              class="sc-AxheI dAvOVz"
            >
              sdsdsD
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
